java/core/src/main/java/co/worklytics/psoxy/gateway/impl/CommonRequestHandler.java

```diff
@@ -207,19 +205,20 @@ boolean isSuccessFamily(int statusCode) {
 
     @SneakyThrows
     public URL buildTarget(HttpEventRequest request) {
+        // contents may come encoded. It should respect url as it comes.
+        // Construct URL directly concatenating instead of URIBuilder as it may re-encode.
         URIBuilder uriBuilder = new URIBuilder();
         uriBuilder.setScheme("https");
         uriBuilder.setHost(config.getConfigPropertyOrError(ProxyConfigProperty.TARGET_HOST));
-        // URL comes encoded, decode it prior to perform call to API origin to avoid double encoding issues
-        // per documentation, both setPath & setCustomQuery:
-        // https://javadoc.io/doc/org.apache.httpcomponents/httpclient/latest/org/apache/http/client/utils/URIBuilder.html#setPath(java.lang.String)
-        // The value is expected to be unescaped and may contain non ASCII characters.
-        uriBuilder.setPath(URLDecoder.decode(request.getPath(), StandardCharsets.UTF_8));
+        URL hostURL = uriBuilder.build().toURL();
+        String hostPlusPath =
```

Comment:
follow same approach as in https://github.com/Worklytics/evalengin/pull/4962/files proxy utils, concatenate URLs to not involve extra encodings.